src/cogent3/app/__init__.py

```diff
@@ -18,9 +18,31 @@
 
 def _get_app_attr(name, obj, mod, is_composable):
```

Comment:
I don't think you should need to make changes to this function. See other comments.

Reply:
Thanks so much for your comments Gavin! 

Regarding FrozenSet issue, I noticed all the raw Strings would be converted into frozen sets in the inherited class `ComposableType`, do we still need make the attributes to be frozen sets in composable apps?

Reply:
You're right, it is not necessary.